hs_tools_resource/utils.py

```diff
@@ -58,3 +62,48 @@ def get_SupportedSharingStatus_choices():
             ['Discoverable', 'Discoverable'],
             ['Private', 'Private'],
             ]
+
+
+def do_work_when_launching_app_as_needed(app_shortkey, res_shortkey, user):
+    """
+    check whether there are extra work needed to be done when the app is launched.
+    :param app_shortkey: shortkey of the app tool resource to be launched
+    :param res_shortkey: shortkey of the resource launching the app tool resource
+    :param user: requesting user
+    :return: error message if an exception is raised; otherwise, return None
+    """
+
+    # When app resource has iRODS federation target path and target resource defined as
+    # extended metadata, the resource needs to be pushed to the specified iRODS federation path
+    # in the specified iRODS storage resource.
+    try:
+        res = get_resource_by_shortkey(res_shortkey)
+        app_res = get_resource_by_shortkey(app_shortkey)
+
+        # check whether irods_path_key and irods_resc_key are added as extended metadata of the
+        # app tool resource, and if they are, pass them out as context variables for app launcher
+        # to push resource to federated iRODS path accordingly
+        irods_path_key = 'irods_federation_target_path'
+        irods_resc_key = 'irods_federation_target_resource'
+        filterd_app_obj = BaseResource.objects.filter(short_id=app_shortkey).filter(
```

Comment:
This could be pulled out of the try block and checked at the beginning.  That way we're not getting the resource and app_resource every time this is called.

Reply:
Good catch, will do.